src/components/filter/VsFilter.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+    <div
+        class="vs-filter"
+        data-test="vs-filter"
+    >
+        <VsButton
+            class="vs-filter__toggle"
+            icon="filters"
+            id="toggle-btn"
+            ref="btnShow"
+            variant="secondary"
+            @click="emitter.emit('showModal', props.filterId)"
+        >
+            {{ props.filterButtonText }}
+        </VsButton>
+
+        <VsModal
```

Comment:
Is there a fall back for users with no javascript?

Reply:
This feels like the sort of thing where we might just have to hide the whole component in no-js, but I'm not sure if that's up to the spec

Reply:
You can use the VsWarning component for fallback messages